### ISSUE TITLE
Free-free and free-bound continuum does not match IDL output

### ISSUE THREAD

**Issue by wtbarnes** [chianti-atomic/ChiantiPy#108]
The free-free and free-bound emissivity (calculated using `continuum.freeFree()` and `continuum.freeBound()`) does not match the equivalent results in IDL. Relevant plots and code snippets in both IDL and Python are included below. 

In both cases, the emission is calculated over 25-414 angstroms in wavelength and 10^5 K to 10^8 K in temperature.

## Free-free
![image](https://cloud.githubusercontent.com/assets/8676141/24011450/d0eb86e0-0a48-11e7-9702-f0f594e223c3.png)
![image](https://cloud.githubusercontent.com/assets/8676141/24011457/d9e7c6d2-0a48-11e7-8ad9-dbd8ae6835c2.png)

## Free-bound
![image](https://cloud.githubusercontent.com/assets/8676141/24011544/18dc1fd2-0a49-11e7-8f54-5bea27ca09b0.png)
![image](https://cloud.githubusercontent.com/assets/8676141/24011565/20881588-0a49-11e7-8430-f9fdeb4eb996.png)

## Code
Script to produce IDL results,
```IDL
; set common block
common abund, abund_ref, ioneq, ioneq_logt, ioneq_ref

; read abundance and ionization equilibrium
abundfile="/usr/local/ssw/packages/chianti/dbase/abundance/sun_coronal_1992_feldman.abund"
read_abund, abundfile, abund, abund_ref
ioneqfile="/usr/local/ssw/packages/chianti/dbase/ioneq/chianti.ioneq"
read_ioneq, ioneqfile, ioneq_logt, ioneq, ioneq_ref

; set temperature and wavelength
pressure = 1.e15
temperature = float(ioneq_logt)
wavelength = findgen(413-24) + 25
edensity = pressure/(10.^temperature)

; proton-to-electron density ratio
common proton, pstr, pe_ratio
pe_ratio=proton_dens(temperature)

; calculate free-free
freefree, 10.^temperature, wavelength, ff, /photons, /no_setup
; calculate free-bound
freebound, 10.^temperature, wavelength, fb, /photons, /no_setup

; make into normal units
tp = tp/1d40
ff = ff/1d40
fb = fb/1d40
```

Script to produce the Python results,
```Python
import ChiantiPy.tools.data as ch_data
import ChiantiPy.core as ch
import astropy.constants as const
import astropy.units as u

ch_data.Defaults['abundfile'] = 'sun_coronal_1992_feldman'
ch_data.Defaults['ioneqfile'] = 'chianti'

temperature = 10.**np.arange(5,8,61)
wavelength = np.arange(25,414,1.)
# need to divide by photon energy because ChiantiPy continuum outputs results in 
# erg cm^-3 s^-1 sr^-1 angstrom^-1
photon_energy = ((const.h*const.c).to(u.erg*u.angstrom)/(wavelength*u.angstrom)).value

freefree = np.zeros((len(temperature),len(wavelength)))
freebound = np.zeros((len(temperature),len(wavelength)))

for ion_name in ch_data.MasterList:
    tmp_continuum = ch.continuum(ion_name,temperature=temperature)
    if tmp_continuum.Ion > 1:
        tmp_continuum.freeFree(wavelength)
        if 'rate' in tmp_continuum.FreeFree:
            freefree += tmp_continuum.FreeFree['rate']/photon_energy
        tmp_continuum.freeBound(wavelength)
        if 'rate' in tmp_continuum.FreeBound:
            freebound += tmp_continuum.FreeBound['rate']/photon_energy
```

**Comment by kdere** [comment]
One of the problems with the free-free is the use of the non-relativistic gaunt factors.  It is possible that it is that simple except the interpolation of the relativistic/Itoh values is complicated so I have not done it.

There is no simple explanation for the free-bound except that it is a mess.  At this point, I am not sure what the easiest way to chase this down.  Perhaps it can be done a single ion at a time.  It may be the free-bound gaunt factor.

Also, the proton density should not enter into any of this.

**Comment by wtbarnes** [comment]
Ok. I just wanted to make sure I was not making a simple mistake in interpreting the output, either on the IDL or Python end. 

I will take a look into both the free-free and the free-bound methods then. I'm currently trying to use both of these to include the continuum contribution in the calculation of the AIA response functions in SunPy so I have a stake in getting these right!

The calculation for both methods seems pretty clearly laid out in both of the IDL methods. I will try to follow those and the relevant papers and see where I get.

**Comment by wtbarnes** [comment]
One question regarding both the Python and IDL free-free methods:

In both the relativistic and non-relativistic formulas of Itoh et al. (2000) (their Eqs. 1 and 7, respectively), there is a Z_j^2 term and they denote that Z_j is the charge of the ion (e.g. 17 for Fe XVIII). However, in both the CHIANTI IDL routines (L350 of itoh.pro) and the [ChiantiPy routines](https://github.com/chianti-atomic/ChiantiPy/blob/master/ChiantiPy/core/Continuum.py#L729), this Z_j^2 term seems to be written as Z^2 (e.g. 26 for any ion of Fe). Perhaps I am just missing something here...

This of course does not account for any differences in the IDL vs. Python results.

**Comment by kdere** [comment]
I really have not looked at this for a long time and I rarely do anything in IDL anymore.
Peter Young (pyoung9@gmu.edu) redid the Itoh a while ago.  The Itoh calculations do not include the full area of temperature and wavelength that we typically use.  He then inserted the Sutherland values for the remaining area.  It would be a good idea to talk to him.  For the Itoh gaunt factors, I was planning to eventually use the scipy.spatial routines, probably Delaunay for doing this but did not get around to it as it is somewhat complicated.
For the free-bound, I think we use the same gaunt factors.  The best approach would be to chase things down ion by ion.  It probably would not take many to find out what is happening

**Comment by kdere** [comment]
as for the question about Z_j^2 vs Z^2, I would ask Peter Young.  Generally Z is the nuclear charge and for relativistic calculations I would not think the electrons present would change this.

**Comment by wtbarnes** [comment]
Thanks for putting me in touch with Peter.

Doesn't the free-free ChiantiPy continuum include the gaunt factors of Itoh ([starting at this line](https://github.com/chianti-atomic/ChiantiPy/blob/master/ChiantiPy/core/Continuum.py#L680))? It looks like you've implemented roughly the same correction as the IDL code: calculate Itoh gaunt factors and then fill in with Sutherland where the Itoh values are not valid.

Disregard the last comment re: Z_j vs Z. I think I was just misinterpreting how the expression for the free-free emission is written in a few different papers.

**Comment by kdere** [comment]
I had forgotten about that.  It does take the Sutherland if it seems to be OK and then, if not, uses the Itoh.  The Itoh factors are probably(?) only rarely used.

**Comment by wtbarnes** [comment]
As an update, here are the corrected free-free and free-bound emission as calculated by ChiantiPy. They compare quite well to the IDL results (as seen above),
![image](https://cloud.githubusercontent.com/assets/8676141/25051644/5222d476-2113-11e7-9860-8f7a0d4980ed.png)

![image](https://cloud.githubusercontent.com/assets/8676141/25051633/45267084-2113-11e7-91d2-909bee724274.png)

**Comment by kdere** [comment]
looks much better!
I am just trying to clarify a few thing before doing the PR.

**Comment by wtbarnes** [comment]
Closed by #114
